src/test/rgw/test_redis_lock.lua

```diff
@@ -0,0 +1,72 @@
+#!lua name=liblock
+
+--- Linux Error codes
+local lerrorCodes = {
+    EPERM = 1,
+    ENOENT = 2,
+    EBUSY = 16,
+    EEXIST = 17
+}
+
+local function myfunction(keys, args)
+    return 'Hello From Lua!'
+end
+
+redis.register_function('myfunction', myfunction)
+
+--- Acquire a lock on a resource.
+--- It sets a key with a cookie value if the key does not exist.
+--- If the key exists and the value is same as cookie, it extends the lock.
+--- If the key exists and the value is different from cookie, it fails.  
+---@param keys table A single element list - lock name
+---@param args table A two-element list - cookie and timeout
+local function lock(keys, args)
+    local name = keys[1]
+    local cookie = args[1]
+    local timeout = args[2]
+    local lock_status = assert_lock(keys, args)
+    if lock_status == 0 then
+        redis.call('PEXPIRE', name, timeout)
+        return 0
+    elseif lock_status == -lerrorCodes.ENOENT then
+        redis.call('SET', name, cookie, 'PX', timeout)
+        return 0
+    end
+    return lock_status
+end
+
+--- Release the lock on a resource.
+--- It deletes the key if the value matches the cookie.
+---@param keys table A single element list - lock name
+---@param args table A single-element list - cookie
+local function unlock(keys, args)
+    local name = keys[1]
+    local cookie = args[1]
+    local existing_cookie = redis.call('GET', name)
+    if existing_cookie == cookie then
+        return redis.call('DEL', name)
+    end
+    return 0
```

Comment:
if we are trying to unlock someone else's lock, shouldn't we return `-EBUSY`?

Reply:
Yup, it does return EBUSY in that case.